src/index.ts

```diff
@@ -64,24 +54,42 @@ export type JsonSDFGEdge = {
     y?: number,
 };
 
-export type JsonSDFGNode = {
+export interface JsonSDFGNode {
     attributes: any,
     id: number,
     label: string,
     scope_entry: string | null,
     scope_exit: string | null,
     type: string,
-};
+}
 
-export type JsonSDFGState = {
+export interface JsonSDFGBlock {
     attributes: any,
     collapsed: boolean,
     edges: JsonSDFGEdge[],
+    nodes: (JsonSDFGBlock | JsonSDFGNode)[],
     id: number,
     label: string,
-    nodes: JsonSDFGNode[],
+    type: string,
+}
+
+export interface JsonSDFGControlBlock extends JsonSDFGBlock {
+    nodes: JsonSDFGBlock[],
+}
+
+export interface JsonSDFGState extends JsonSDFGBlock {
     scope_dict: any,
+    nodes: JsonSDFGNode[],
+}
+
+export type JsonSDFG = {
```

Comment:
So why did this move? can you change `any[]` in nodes to blocks and edges to JsonSDFGEdge now?

Reply:
I am in the process of doing that, but it will most likely have to be a separate PR. It's a larger-than-expected type system refactor, but I am planning on doing a type system and code style cleanup towards the end of the year (should enable easier contribution by third parties) - this will most likely happen there.

A reason for the larger refactoring in types is that the current interactions with the graph still assume all SDFG states are direct descendants of the SDFG object (i.e., in `sdfg.nodes`). This needs to be adapted first, but the concrete solution is not fixed yet and depends on the outcome of https://github.com/spcl/dace/pull/1356.